os/arch/arm/src/amebasmart/amebasmart_usb.c

```diff
@@ -0,0 +1,803 @@
+
+static void rtl8730e_log_usb_detach(struct uart_dev_s *dev)
+{
+
+}
+
+/****************************************************************************
+ * Name: usb_ioctl
+ *
+ * Description:
+ *   All ioctl calls will be routed through this method
+ *
+ ****************************************************************************/
+
+static int rtl8730e_log_usb_ioctl(FAR struct uart_dev_s *dev, int cmd, unsigned long arg)
+{
+	int ret = OK;
+	return ret;
+}
+
+/****************************************************************************
+ * Name: usb_receive
+ *
+ * Description:
+ *   Called (usually) from the interrupt level to receive one
+ *   character from the UART.  Error bits associated with the
+ *   receipt are provided in the return 'status'.
+ *
+ ****************************************************************************/
+
+static int rtl8730e_log_usb_receive(struct uart_dev_s *dev, unsigned int *status)
+{
+	struct rtl8730e_up_dev_s *priv = (struct rtl8730e_up_dev_s *)dev->priv;
+	uint8_t rxd;
+	
+	DEBUGASSERT(priv);
+	while(!received_flag){
+
+	}
+	rxd = receive_data;
+	*status = rxd;
+	received_flag = 0;
+	return rxd & 0xff;
+}
+
+
+/****************************************************************************
+ * Name: usb_rxint
+ *
+ * Description:
+ *   Call to enable or disable RX interrupts
+ *
+ ****************************************************************************/
+static void rtl8730e_log_usb_rxint(struct uart_dev_s *dev, bool enable)
+{
+
+}
+
+/****************************************************************************
+ * Name: usb_rxavailable
+ *
+ * Description:
+ *   Return true if the receive fifo is not empty
+ *
+ ****************************************************************************/
+
+static bool rtl8730e_log_usb_rxavailable(struct uart_dev_s *dev)
+{
+	if(received_flag) {
+		return TRUE;
+	} else {
+		return FALSE;
+	}
+}
+
+/****************************************************************************
+ * Name: usb_send
+ *
+ * Description:
+ *   This method will send one byte on the USB
+ *
+ ****************************************************************************/
+static void rtl8730e_log_usb_send(struct uart_dev_s *dev, int ch)
+{
+	struct rtl8730e_up_dev_s *priv = (struct rtl8730e_up_dev_s *)dev->priv;
+	DEBUGASSERT(priv);
+	int fd;
+	
+	usbd_cdc_acm_transmit(&ch, 1);
+	while (sem_wait(&g_usbdev.txsem) != 0) {
+		/* The only case that an error should occur here is if the wait was awakened
+		 * by a signal.
+		 */
+		ASSERT(errno == EINTR);
+	}
+	priv->tx_level--;
+
+}
+
+
+static void usb_xmitchars(FAR uart_dev_t *dev)
+{
+	struct rtl8730e_up_dev_s *priv = (struct rtl8730e_up_dev_s *)dev->priv;
+	DEBUGASSERT(priv);
+	uint16_t nbytes = 0;
+	uint8_t *buffer = NULL;
+	buffer = (uint8_t *)rtw_zmalloc(CONFIG_CDC_ACM_BULK_IN_XFER_SIZE);
+	if (buffer == NULL) {
+		lldbg("malloc failed\n");
+	}
+
+	irqstate_t flags = enter_critical_section();
```

Comment:
leave_critical_section is in the CONFIG_SMP conditional. But here is no. Need to be in the same conditional.

Reply:
added CONFIG_SMP